build_defs/souffle.bzl

```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 #-----------------------------------------------------------------------------
 
-def souffle_cc_library(name, src, included_dl_scripts=[], visibility = None):
+def souffle_cc_library(name, src, all_principals_own_all_tags = False, included_dl_scripts=[], visibility = None):
```

Comment:
Can we restrict `all_principals_own_all_tags` to `test_only` targets somehow? Or, is it already the case?

Reply:
That is worth looking into. It's currently the case that `all_principals_own_all_tags` is only used in test contexts, but it is not the case that there is any automatic method restricting that to be the case.

Reply:
I think that, while something like `additional_souffle_flags` might be valuable for other flags, this behavior deserves its own name. This behavior is very test-only, and so I'm going to be tagging the generated rules as `testonly` if this flag is set. Because this will both add the flag and alter the rule, it should have one name that does both in concert to prevent the behavior from accidentally getting sheared someplace.

Reply:
If `all_principals_own_all_tags` is `True`, all rules generated by the macro will be `testonly`, which should guard against this leaking into non-test builds.